src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -137,4 +142,56 @@
         }
         return new ProjectName(trimmedName);
     }
+
+    /**
+     * Parses a String into a description
+     *
+     * @param description
+     * @return description.
+     * @throws ParseException exception related.
+     */
+    public static String parseDescription(String description) throws ParseException {
+        requireNonNull(description);
+        String trimmedDescription = description.trim();
+        if (!trimmedDescription.matches("[\\p{Alnum}][\\p{Alnum} ]*")) {
+            throw new ParseException(Event.MESSAGE_DESCRIPTION_CONSTRAINTS);
+        }
+        return trimmedDescription;
+    }
+
+    /**
+     * Parses a String into an interval enum.
+     *
+     * @param interval
+     * @return Interval enum.
+     * @throws ParseException exception related.
+     */
+    public static Interval parseInterval(String interval) throws ParseException {
+        requireNonNull(interval);
+        String trimmedInterval = interval.trim();
+        try {
+            Interval.valueOf(trimmedInterval);
+        } catch (IllegalArgumentException iae) {
+            throw new ParseException(Event.MESSAGE_INTERVAL_CONSTRAINTS);
+        }
+        return Interval.valueOf(trimmedInterval);
```

Comment:
put the return inside the try catch block. try { return Interval....}. In general, good to avoid returning outside the try catch block unless necessary as it's easier to read (usually if you have to it might be better to abstract out the try catch block in a method).

Reply:
resolved